internal/federationdomain/resolvedprovider/resolvedgithub/resolved_github_provider.go

```diff
@@ -68,8 +70,17 @@ func (p *FederationDomainResolvedGitHubIdentityProvider) UpstreamAuthorizeRedire
 	state *resolvedprovider.UpstreamAuthorizeRequestState,
 	downstreamIssuerURL string,
 ) (string, error) {
-	fmt.Printf("GithubResolvedIdentityProvider ~ UpstreamAuthorizeRedirectURL() called with state: %#v, downstreamIssuerURL %s", state, downstreamIssuerURL)
-	return "", errors.New("function UpstreamAuthorizeRedirectURL not yet implemented for GitHub IDP")
+	upstreamOAuthConfig := oauth2.Config{
+		ClientID: p.Provider.GetClientID(),
+		Endpoint: oauth2.Endpoint{
+			AuthURL: p.Provider.GetAuthorizationURL(),
```

Comment:
Is this going to use the right hostname and path when the user has specified a custom GitHub hostname on their GitHubIdentityProvider? Note that the paths are different for GitHub Enterprise.

Reply:
The `authorizationURL` is populated by `github_upstream_watcher.go`, and it looks from the tests that it has the form: `"https://<spec.githubAPI.host>/login/oauth/authorize"`

https://github.com/vmware-tanzu/pinniped/blob/github_identity_provider/internal/controller/supervisorconfig/githubupstreamwatcher/github_upstream_watcher_test.go

Reply:
That should be the path for github.com, I'll take a quick look at GitHub enterprise.